s3po/backends/s3.py

```diff
@@ -93,3 +93,9 @@ def func():
             multi.upload_part_from_file(StringIO(data), count)
             multi.complete_upload()
             return True
+
+    def list(self, bucket, prefix=None, delimiter=None, headers=None):
```

Comment:
Do you think it's insufficient to simply have this be a generator? If we're going to go that route, `itertools.irange` is the tool you're looking for, and it could probably fit one level up at `Connection`.